ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/ChainSync/Client/Jumping.hs

```diff
@@ -391,7 +391,8 @@ onRollForward context point = whenEnabled context () $
     Disengaged -> pure ()
     Jumper{} -> pure ()
     Dynamo lastJumpSlot
-      | pointSlot point >= ((jumpSize context +) <$> lastJumpSlot) -> do
+      | let jumpBoundaryPlus1 = jumpSize context + succWithOrigin lastJumpSlot
+      , succWithOrigin (pointSlot point) >= jumpBoundaryPlus1 -> do
```

Comment:
This fixes the weird interaction of `+`, `<$>` and `WithOrigin`; nice!

However, when the new point is exactly jumpSize away from the previous jump point (ie. in the equality case here), we trigger a jump to the tip of the candidate fragment, even though the new point would be a “better” jump point.

For instance:
```
jumpSize = 3

gtBlockTree:                             
  slots:    0  1  2  3  4  5  6  7  8  9 
  trunk:  ──1─────2──3─────4──5─────6──7 
``` 

We could hope to jump from G to 2-2, then 4-5, then 6-8. Instead, 2-2 will be the first point to pass this condition (see below) and we will trigger a jump to 0-1. Then, 3-3 will be the next point passing the condition again, triggering us to jump to 2-2. etc.

```haskell
let jumpBoundaryPlus1 = jumpSize context + succWithOrigin lastJumpSlot -- = 3 + succWithOrigin G == 3
succWithOrigin (pointSlot point) >= jumpBoundaryPlus1 -- succWithOrigin 2 >= 3 == True
```

This is probably not a very big deal on big jumpSizes (eg. half main net's genesis window) but it is still counter-intuitive. Should we put greater than instead? Do you see any shortcomings to this solution?

Reply:
> Do you see any shortcomings to this solution?

No shortcomings. We could try that.